### DISCUSSION TITLE
Missing/Not added report feature in lazer's chat.

### DISCUSSION THREAD

**Discussion by msmcculley** [ppy/osu#15213] [Q&A] [answered]
Hey all, 

First and foremost, if this is a duplicate post then I apologize I couldn't find one that was similar or not. 
Second, if it just hasn't been added because it is on the back-burner of the development team because it is just a chat function then I completely understand. 

Now, to the actual problem/content: 
Seems that in comparison to stable/anything not lazer there is something other than the !report <username> <reason>, whether it be the right click of the mouse and either pressing 6 or clicking the report button and then filling out the reason(s) there for either simplicity or nuance's sake. 

However, there is no such feature like that in lazer with the screenshots been taken 10/21/2021 and show no report feature if you are looking at the webpage version of the userpage or if there would be an in-client report feature (besides the command). 

![bTVZvNg](https://user-images.githubusercontent.com/90729492/143930253-a25f30b1-2665-4efd-8c48-e247727b60d7.jpeg)

![7dSO5cD](https://user-images.githubusercontent.com/90729492/143930302-1db257a2-d657-4bcf-8d03-cb6d2b13f06f.png)

Edit is to talk about some other issues that I have found after I initially posted this: 
Tab no longer finishes the name of a user (regardless if they are on IRC or in-client) 

I originally posted something about the difference between /help and !help where /help does not show any indication that "!help" is a command. I notice now that they are two different help commands for two different command lists. So, that is also being removed from the OP. 

![XQ9ZMqc](https://user-images.githubusercontent.com/90729492/143930465-b89bc3a8-610e-4e0b-b371-e124e6392b10.jpeg)

 this screenshot shows when I typed "/help" and it gives me the error of an unknown command, but when I type "!help" bancho pastes the commands. *
*Edit #2 was just for fixing the images*

**Comment by Nukrid** [comment]
It is a good idea to add the button, it is the method most people use to actually report misbehavior. Making things simpler is always a good thing for the userbase.

**Comment by msmcculley** [comment]
Now that chat elements are being implemented, I was wondering if this would be the next step?

**Comment by msmcculley** [comment]
Hello! (Once again!)

I was wondering if I could get some help on this. 

I currently have the button selectable and (seemingly will do an action) but as I am new to programming I am not exactly sure how to proceed with the API/Logic associated with the action of the report button

More specifically I am using the comment report system because it holds the same functions that are talked about in the initial discussion above. Plus there is no documentation on something like this that I can find (but I mightve not looked hard enough besides: (ReportCommentPopover.cs),(CommentReportReason.cs), and CommentReportButton.cs), which I have zero knowledge in applying which parameters to the `Action = () =>` portion of the code.  

![r7CrUm4](https://user-images.githubusercontent.com/90729492/207508346-2e4af122-b999-4a01-b6a4-d9548c988987.png)

![](https://i.imgur.com/GKbf1Ra.png)

**Reply by cdwcgt** [reply]
just send !report {username} {reason} is fine

**Reply by cdwcgt** [reply]
And I have a feeling your code version might be outdated

**Comment by cdwcgt** [comment, marked as answer]
resolved in https://github.com/ppy/osu/pull/21736
